.github/workflows/release.yml

```diff
@@ -0,0 +1,112 @@
+name: Build and release Unreal Plugin  🎉 
+
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  packages: write
+
+env:
+  CPP_SDK_TAG: "2.3.3"
+  CPP_SDK_DESTINATION: "DolbyIO"
+  CPP_SDK_URL: "https://github.com/DolbyIO/comms-sdk-cpp/releases/download"
+  UNREAL_PLUGIN_OUTPUT: "Build/DolbyIO"
+  RETENTION: 7
+
+jobs:
+  build-macos:
+    strategy:
+      fail-fast: false
+      matrix:
+        version: ["4.27", "5.0", "5.1"]
+    runs-on: ["self-hosted", "X64", "macOS"]
+    env:
+      BUILD_PLATFORM: 'macos64'
+      GH_TOKEN: ${{ github.token }}
+
+    steps:
+      - name: Code checkout
+        uses: actions/checkout@v3
+      - name: Download C++ SDK release asset
+        uses: ./.github/actions/download_cpp/bash
+        with:
+          url: ${{ env.CPP_SDK_URL }}
+          tag: ${{ env.CPP_SDK_TAG }}
+          asset_name: "cppsdk-${{ env.CPP_SDK_TAG }}-${{ env.BUILD_PLATFORM }}.zip"
+      - name: Remove symlinks from cpp sdk libs
+        env:
+          LIB_PATH_ARM: "${{ github.workspace }}/${{ env.CPP_SDK_DESTINATION }}/sdk-release-arm/lib"
+          LIB_PATH_X86: "${{ github.workspace }}/${{ env.CPP_SDK_DESTINATION }}/sdk-release-x86/lib"
+        run: |
+          echo "Removing symlinks from ${LIB_PATH_ARM}"
+          rm ${LIB_PATH_ARM}/libav*.dylib
+          echo "Removing symlinks from ${LIB_PATH_X86}"
+          rm ${LIB_PATH_X86}/libav*.dylib
+      - name: Run Build
+        run: |
+          /Users/Shared/Epic\ Games/UE_${{ matrix.version }}/Engine/Build/BatchFiles/RunUAT.sh BuildPlugin \
+          -Rocket \
+          -Plugin=$(pwd)/DolbyIO/DolbyIO.uplugin \
+          -TargetPlatforms=Mac \
+          -Package=$(pwd)/${{ env.UNREAL_PLUGIN_OUTPUT }} \
+          -StrictIncludes
+      - name: Sign DolbyIO Unreal Plugin
+        env:
+          DOLBYIO_KEYCHAIN_PASSWORD: ${{ secrets.VOXEET_KEYCHAIN_PASSWORD }}
+        uses: ./.github/actions/sign/mac
+        with:
+          path: ${{ env.UNREAL_PLUGIN_OUTPUT }}
+          platform: 'macos'
+      - name: Zip of assets of DolbyIO Unreal Plugin
+        working-directory: ${{ env.UNREAL_PLUGIN_OUTPUT }}
+        run: |
+          zip -rq DolbyIO_UnrealPlugin_${{ matrix.version }}_${{ env.BUILD_PLATFORM }}.zip .
+      - name: Send assets of DolbyIO Unreal Plugin 🎉
+        run: |
+          ./.github/actions/upload_assets/upload_release_asset.sh ${{ env.UNREAL_PLUGIN_OUTPUT }}/DolbyIO_UnrealPlugin_${{ matrix.version }}_${{ env.BUILD_PLATFORM }}.zip
```

Comment:
Zip file name is wrong. E.g. Build.DolbyIO.DolbyIO_UnrealPlugin_4.27_macos64.zip. It should be DolbyIO_UnrealPlugin_4.27_macos64.zip and after unzipping at lest the same. Perfect situation would be when after unzipping we could have just DolbyIO.